packages/webpack-plugin/src/plugin.ts

```diff
@@ -0,0 +1,463 @@
+                module.buildMeta.stylable.depth = calcDepth(module, moduleGraph);
+            }
+        });
+
+        compilation.hooks.afterChunks.tap(StylableWebpackPlugin.name, () => {
+            const optimizer = this.stylable.optimizer!;
+            const optimizeOptions = this.options.optimize;
+            const sortedModules = getSortedModules(stylableModules);
+
+            const {
+                usageMapping,
+                namespaceMapping,
+                namespaceToFileMapping,
+            } = createOptimizationMapping(sortedModules, optimizer);
+
+            if (!this.options.unsafeMuteDiagnostics.DUPLICATE_MODULE_NAMESPACE) {
+                reportNamespaceCollision(namespaceToFileMapping, compilation.errors);
+            }
+
+            for (const module of sortedModules) {
+                const buildMeta = getStylableBuildMeta(module);
+                const { css, exports, globals } = buildMeta;
+
+                const ast = parse(css);
+
+                optimizer.optimizeAst(
+                    optimizeOptions,
+                    ast,
+                    usageMapping,
+                    this.stylable.delimiter,
+                    exports,
+                    globals
+                );
+
+                buildMeta.css = optimizeOptions.minify
+                    ? optimizer.minifyCSS(ast.toString())
+                    : ast.toString();
+
+                if (optimizeOptions.shortNamespaces) {
+                    buildMeta.namespace = namespaceMapping[buildMeta.namespace];
+                }
+            }
+        });
+    }
+    private chunksIntegration(
+        compilation: Compilation,
+        staticPublicPath: string,
+        stylableModules: Set<NormalModule>,
+        assetsModules: Map<string, NormalModule>
+    ) {
+        const { runtimeTemplate } = compilation;
+
+        /**
+         * As a work around unknown behavior
+         * if this plugin will run inside a child compilation we do not emit css assets
+         */
+        if (!compilation.compiler.isChild()) {
+            if (this.options.cssInjection === 'css') {
+                compilation.hooks.processAssets.tap(
+                    {
+                        name: StylableWebpackPlugin.name,
+                        stage: Compilation.PROCESS_ASSETS_STAGE_DERIVED,
+                    },
+                    () => {
+                        const cssSource = createStaticCSS(
+                            staticPublicPath,
+                            stylableModules,
+                            assetsModules,
+
+                            compilation.chunkGraph!,
+                            compilation.moduleGraph,
+                            'CSS' /*runtime*/,
+                            compilation.runtimeTemplate,
+                            compilation.dependencyTemplates
+                        ).join('\n');
```

Comment:
not os.EOL?

Reply:
I was not sure about is previously and still not sure....why should I respect the os \n here?